src/java_bytecode/jar_file.h

```diff
@@ -15,27 +15,27 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <string>
 #include <vector>
 #include <map>
+#include <regex>
+#include <util/message.h>
 
-class jar_filet
+class jar_filet:public messaget
```

Comment:
this trickles down, at least that's what I intended; the class loader sets the message handler of the `jar_poolt`
```
  void set_java_cp_include_files(std::string &java_cp_include_files)
  {
    jar_pool.set_java_cp_include_files(java_cp_include_files);
    jar_pool.set_message_handler(get_message_handler());
  }
```
and the `jar_poolt` sets the handler of the `jar_filet`
```
  jar_filet &operator()(const std::string &file_name)
  {
    if(java_cp_include_files.empty())
      throw "class regexp cannot be empty";
    file_mapt::iterator it=file_map.find(file_name);
    if(it==file_map.end())
    {
      jar_filet &jar_file=file_map[file_name];
      jar_file.set_message_handler(get_message_handler());
      jar_file.open(java_cp_include_files, file_name);
      return jar_file;
    }
    else
      return file_map[file_name];
  }
```
Out of laziness, I do not put it in the constructor. Also having an explicit constructor for jar_poolt raises errors as the implicit default constructor would disappear, which is used in the code. Therefore this approach has the least impact, and jar_filet/jar_poolt is only used here.